### ISSUE TITLE
Hot-reload dependencies on file changes

### ISSUE THREAD

**Issue by MartinBernstorff** [marimo-team/marimo#998]
### Description

I am hoping to use Marimo as my "main" script, and develop more complicated logic in an accompanying Python library. To do this most effectively, I need some mechanism for re-loading imports.

In Jupyter, I would use something like https://ipython.readthedocs.io/en/stable/config/extensions/autoreload.html

### Suggested solution

Enable import blocks to be re-loaded on dependent file change.

**Comment by mscolnick** [comment]
This would be a very neat feature, especially when its fully reactive. In the meantime, you can manually reload by including this snippet and re-running your cell:

```python
import mymodule
import importlib
importlib.reload(mymodule)
```
https://docs.marimo.io/faq.html#faq-reload

**Comment by astrowonk** [comment]
Yes, +1 to this. Without jupyter-like `autoreload` functionality (and at a minimum, being able to reimport without importlib workraounds as mentioned in #292), marimo struggles as a jupyter-replacement.

Like @MartinBernstorff, I keep all my main functions and classes in other python files and import them into my notebooks for visualization, testing, etc - making changes to the .py files in VS Code, then testing them out in the notebook.

While Jupyter with `autorleload` can detect changes to existing objects I've imported, marimo won't even detect when I add a new function to a file: `from myfile import newfunc` fails if I add `newfunc` to `myfile.py` after starting marimo.

**Comment by akshayka** [comment]
@astrowonk , @MartinBernstorff : PR #1092 adds hot-reloading in the style of IPython's `%autoreload 2`. It can be enabled via the notebook config menu (gear icon, top right of a notebook, under User Config > Runtime Settings).

This will be available in the next release (by tomorrow).

**Comment by astrowonk** [comment]
Y'all move at blazing speed! Thanks!

**Comment by akshayka** [comment]
> Y'all move at blazing speed! Thanks!

Yep, thanks for the nudge! It's now released, version 0.3.11.

**Comment by MartinBernstorff** [comment]
Awesome work guys! 

Reactive reloading would be a dream come true; it would solve so much friction for me. Have added an issue.
